bundles/binding/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/LcnBindingProvider.java

```diff
@@ -0,0 +1,194 @@
+				if (openHabMatcher.matches()) {
+					openHabCmd = openHabMatcher.group(1);
+					connId = openHabMatcher.group(2);
+					lcnTarget = openHabMatcher.group(3);	
+				} else if (pureMatcher.matches()) {
+					connId = pureMatcher.group(1);
+					lcnTarget = pureMatcher.group(2);
+				} else
+					throw new BindingConfigParseException("Invalid binding configuration for " + binding + "!");
+				String lcnShort = resolveMappings(lcnTarget, openHabCmd);
+				if (lcnShort == null || lcnShort.equals(openHabCmd))
+					lcnShort = lcnTarget;
+				Command cmd = openHabCmd == null ?
+					TypeParser.parseCommand(new StringItem("").getAcceptedCommandTypes(), "") :
+					openHabCmd.equals("%i") ? new StringType("%i") :
+					TypeParser.parseCommand(item.getAcceptedCommandTypes(), openHabCmd);
+				bc.add(new LcnBindingItem.Binding(cmd, connId, lcnShort));
+			}
+		}
+		// Finished
+		this.addBindingConfig(item, bc);
+		for (LcnAddrMod addr : bc.getRelatedModules()) {
+			HashSet<String> l = this.itemNamesByModulCache.get(addr);
+			if (l == null) {
+				l = new HashSet<String>();
+				this.itemNamesByModulCache.put(addr, l);
+			}
+			l.add(item.getName());
+		}
+	}
+	
+	/** {inheritDoc} */
+	@Override
+	public String getBindingType() {
+		return "lcn";
+	}
+
+	/** {inheritDoc} */
+	@Override
+	public void validateItemType(Item item, String bindingConfig) throws BindingConfigParseException {
+		// Everything is accepted
+	}
+	
+	/**
+	 * Resolves LCN commands (with mappings) to plain commands.
+	 * 
+	 * @param lcnTarget the target or a mapping
+	 * @param openHABcmd the command send by openHAB
+	 * @return the resolved result (can be null)
+	 */
+	private static String resolveMappings(String lcnTarget, String openHABcmd) {
+		String result = null;
+		Matcher matcher = PATTERN_MAPPING.matcher(lcnTarget);
+		if (!matcher.matches())
+			result = lcnTarget;
+		else {
+			matcher.reset();
+			matcher.find();			
+			String s1 = matcher.group(1);
+			String s2 = matcher.group(2);
+			TransformationService transformationService = TransformationHelper.getTransformationService(LcnBindingActivator.getContext(), s1);
+			if (transformationService != null) {
+				try {
+					result = transformationService.transform(s2, openHABcmd);
+				} catch (TransformationException e) {
+					result = lcnTarget;
+				}
+			} else
+				result = lcnTarget;
+		}
+		return result;
+	}
+	
+	/**
+	 * Finds an LCN item by name.
+	 * 
+	 * @param itemName the item's name to search for
+	 * @return the {@link LcnBindingItem} or null
+	 */
+	LcnBindingItem getLcnItemConfig(String itemName) {
+		return (LcnBindingItem)this.bindingConfigs.get(itemName);
+	}
+	
+	/**
+	 * Processes input received from a connection.
+	 * 
+	 * @param pckInput the received input
+	 * @param conn the source {@link Connection}
+	 * @param eventPublisher the {@link EventPublisher} to receive updates
+	 */
+	void processInput(Input pckInput, Connection conn, EventPublisher eventPublisher) {
```

Comment:
Furthermore: the purpose of the Binding(Config)Provider should be exactly this: provide configuration. This event handling should be moved from here (to the LcnBinding?)